ennMethods.rb

```diff
@@ -0,0 +1,101 @@
+# Module
+module Enumerable
+  #my_each: clone of each
+  def my_each
+    for x in self do
+      yield(x)
+    end
+    return self
+  end
+
+  #my_each_with_index: clone of each.with_index
+  def my_each_with_index
+    for i in 0...self.length do
+      yield(self[i], i)
+    end
+    return self
+  end
+
+  #my_select: clone of select
+  def my_select
+    i = 0
+    result = []
+    self.my_each do |x|
+      if yield(x)
+        result[i] = x
+        i += 1
+      end
+    end
+    return result
+  end
+
+  #my_all?: clone of all?
+  def my_all?
+    test = true
+    self.my_each do |x|
+      test = test && yield(x)
+    end
+    return test
+  end
+
+  #my_any?: clone of any?
+  def my_any?
+    test = false
+    self.my_each do |x|
+      test = test || yield(x)
+    end
+    return test
+  end
+
+  #my_none?: clone of none?
+  def my_none?
+    test = true
+    self.my_each do |x|
+      test = test || !yield(x)
+    end
+    return test
+  end
+
+  #my_count: clone of count
+  def my_count
+    counter = 0
+    self.my_each do |x|
+      counter += 1 if yield(x)
+    end
+    return counter
+  end
+
+  #my_map: clone of map
+  def my_map_v1    
+    (0...self.length).each do |i|
+      self[i] = yield(self[i])
+    end
+    return self
+  end
+
+  #my_map_v2: clone of map. Accepts both procedures and blocks
+  def my_map_v2(proc = nil)   
+    (0...self.length).each do |i|
+      self[i] = proc.call self[i] if !proc.nil?
+      self[i] = yield(self[i]) if proc.nil?
+    end
+    return self
+  end
+
+  #my_inject: clone of inject
+  def my_inject
+    result = self[0]
+    (1...self.length).my_each do |i|
+      result = yield(result, self[i])
+    end
+
+    return result
+  end
```

Comment:
Not working
```Ruby
> [5,6,7,8, 9, 10].inject(5) { |sum, n| sum + n }  
 => 50 
> [5,6,7,8, 9, 10].my_inject(5) { |sum, n| sum + n }  
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):32
        1: from ennMethods.rb:86:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
> [5,6,7,8, 9, 10].inject(1) { |product, n| product * n }
 => 151200 
> [5,6,7,8, 9, 10].my_inject(1) { |product, n| product * n }
Traceback (most recent call last):
        3: from /usr/local/rvm/rubies/ruby-2.5.1/bin/irb:11:in `<main>'
        2: from (irb):35
        1: from ennMethods.rb:86:in `my_inject'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

Reply:
Change made. `my_inject` wasn't receiving any parameter as input, it worked with the syntax:
`[5,6,7,8,9,10].my_inject do |sum, n|
  sum+n
end`

Now it accepts one optional parameter, so this would be fixed. Also, it will now deal with ranges too, not only arrays.